packages/sdk-js/src/GrowthBook.ts

```diff
@@ -488,12 +501,46 @@ export class GrowthBook<
 
     // Apply new changes
     if (result.inExperiment) {
-      const undo = this._applyDOMChanges(result.value);
-      if (undo) {
-        this._activeAutoExperiments.set(experiment, {
-          undo,
-          valueHash,
-        });
+      if (result.value.urlRedirect) {
+        const currUrl = new URL(this._getContextUrl());
+        const redirectUrl = new URL(result.value.urlRedirect);
+        mergeUrlSearchParams(currUrl.searchParams, redirectUrl.searchParams);
+
+        const url = experiment.persistQueryString
+          ? redirectUrl.toString()
+          : result.value.urlRedirect;
```

Comment:
This is a little weird, since we're doing the whole merge stuff above, but then only using it if persistQueryString is set.

I would restructure this a bit and do:

```js
const url = experiment.persistQueryString
  // New function, takes in two full URL strings and returns a new merged URL string
  ? mergeQueryStrings(this._getContextUrl(), result.value.urlRedirect)
  : result.value.urlRedirect
```